database/engine/pagecache.c

```diff
@@ -122,6 +122,24 @@ void pg_cache_wait_event_unsafe(struct rrdeng_page_descr *descr)
     --pg_cache_descr->waiters;
 }
 
+/*
+ * The caller must hold page descriptor lock.
+ * The lock will be released and re-acquired. The descriptor is not guaranteed
+ * to exist after this function returns.
+ * Returns UV_ETIMEDOUT if timeout_sec seconds pass.
+ */
+int pg_cache_timedwait_event_unsafe(struct rrdeng_page_descr *descr, uint64_t timeout_sec)
+{
+    int ret;
+    struct page_cache_descr *pg_cache_descr = descr->pg_cache_descr;
+
+    ++pg_cache_descr->waiters;
+    ret = uv_cond_timedwait(&pg_cache_descr->cond, &pg_cache_descr->mutex, timeout_sec * NSEC_PER_SEC);
+    --pg_cache_descr->waiters;
+
+    return ret;
+}
+
```

Comment:
This code does not get called.

Reply:
I know, I was experimenting with it when we discussed about timeouts and I left it there as a library function for future use. We have had other functions like that in the past, that we eventually used.